main/src/com/google/refine/io/ProjectMetadataUtilities.java

```diff
@@ -60,25 +61,35 @@
     public static void save(ProjectMetadata projectMeta, File projectDir) throws IOException  {
         File tempFile = new File(projectDir, "metadata.temp.json");
         saveToFile(projectMeta, tempFile);
+        if (tempFile.length() == 0) {
+            throw new IOException("Failed to save project metadata - keeping backups");
+        }
+
+        // TODO Do we want to make sure we can successfully deserialize the file too?
 
         File file = new File(projectDir, "metadata.json");
         File oldFile = new File(projectDir, "metadata.old.json");
 
-        if (oldFile.exists()) {
-            oldFile.delete();
-        }
-        
         if (file.exists()) {
+            if(file.length() > 0) {
+                if (oldFile.exists()) {
+                    oldFile.delete();
+                }
             file.renameTo(oldFile);
+            } else {
+                file.delete();
+            }
         }
 
         tempFile.renameTo(file);
     }
     
     protected static void saveToFile(ProjectMetadata projectMeta, File metadataFile) throws IOException   {
-        Writer writer = new OutputStreamWriter(new FileOutputStream(metadataFile));
+        Writer writer = new OutputStreamWriter(new FileOutputStream(metadataFile), StandardCharsets.UTF_8);
         try {
             ParsingUtilities.defaultWriter.writeValue(writer, projectMeta);
+            // TODO: Do we want the below instead? Doesn't filter recon candidates
+//            ParsingUtilities.saveWriter.writeValue(writer, projectMeta);
```

Comment:
There is no reason for recon candidates to end up in the project metadata, so both writers are equivalent. Perhaps we could use `saveWriter` because we are *saving* things to disk, but it is a style change.

Reply:
I switched to `saveWriter`